packages/@spectrum-icons/workflow/scripts/generateIcons.js

```diff
@@ -26,7 +26,10 @@ function template(iconName) {
 import {Icon} from '@react-spectrum/icon';
 import React from 'react';
 
-export default function ${iconRename}(props) {
+type IconProps = React.ComponentProps<typeof Icon>;
+type ${iconRename}Props = Omit<IconProps, "children">;
+
+export default function ${iconRename}(props: ${iconRename}Props) {
```

Comment:
One small note, all our workflow icons have the same props, could we generate this type once and import it so that there aren't ~800 types that are all essentially the same except for the name?

Reply:
Yeah, we could probably create a types.ts or so file in the root and import the type from there. Alternatively we could expose the type directly from the icon component so it's easier to create custom icons.

Reply:
> Alternatively we could expose the type directly from the icon component so it's easier to create custom icons.

This sounds like the better way to go :)